crates/oxc_transformer/src/decorator/legacy/mod.rs

```diff
@@ -81,6 +72,22 @@ struct ClassDecoratedData<'a> {
     alias_binding: Option<BoundIdentifier<'a>>,
 }
 
+/// Decoration state for a single class level.
+/// This struct groups together all the state needed while processing decorators for one class,
+/// allowing clean management of nested class scenarios through a stack of these structs.
```

Comment:
AI wrote this comment, right? You can tell by how it's excessively long, and the last 2 lines don't really say anything! 😄

```suggestion
/// Decoration state for a single class.
```

Reply:
Ah, yes. Most of the comments are written by AI, and I tweak them a little bit.